pkg/controller/daemon/daemoncontroller.go

```diff
@@ -994,32 +990,37 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 
 	if ds.DeletionTimestamp != nil || !dsc.expectations.SatisfiedExpectations(dsKey) {
 		// Only update status.
-		return dsc.updateDaemonSetStatus(ds, hash)
+		return dsc.updateDaemonSetStatus(ds, hash, nodeList)
 	}
 
-	err = dsc.manage(ds, hash)
+	err = dsc.manage(ds, hash, nodeList)
 	if err != nil {
 		return err
+	} else {
```

Comment:
@lukaszo - true, the thing is that i need to enqueue + update status after manage call as stated in #47243 and on the next sync start the rolling update. the tricky part is knowing which action to take since current action will depend on the one before.

P.S: since it's a WIP, this PR is not really ready to test (but it should be soon)